### PR TITLE
Better feedback

### PR DESCRIPTION
**Pull request by Siddharth11**
### success alert

feedback

### error alert
error

### CODE REVIEW

**Comment by asanjeevrao** [comment]
-Because success message is being controlled by the URL, after seeing a success message you can navigate forward and back on your browser and see the message again - this shouldn't happen

-If you try to enter a store in a different location (i'm in Vancouver, tried submitting a store in Bangalore), it shows success message and puts the store at position 1 but map changes to Vancouver

image

-I think it'll be a nicer experience to see the map marker as well with the success message, i.e. go back to the home page with it scrolled up so you can see the map and your entry. We can actually replace the current "Help 35,000.." message with this success message 

-How would the share with your friends work with this disappearing flyout? Should we make it stay till the user dismisses it? https://github.com/covid-maps/covid-maps/pull/152

**Comment by Siddharth11** [comment]
> -Because success message is being controlled by the URL, after seeing a success message you can navigate forward and back on your browser and see the message again - this shouldn't happen

@asanjeevrao  to avoid this I can trigger the `history.replace` method (just after determining the right store) to not store the query param in browser history so then if user clicks on back button, he will go to the submit form

**Comment by Siddharth11** [comment]
> -If you try to enter a store in a different location (i'm in Vancouver, tried submitting a store in Bangalore), it shows success message and puts the store at position 1 but map changes to Vancouver

@asanjeevrao this should be fixed now

File: src/Homepage/SearchResults.js [hunk 1]
```diff
@@ -29,9 +39,10 @@ class SearchResults extends React.Component {
   }
 
   render() {
-    const selectedResult = this.props.results.find(result =>
+    const selectedResult = this.props.results.find(result => (
```
Comment (arjun27) [inline src/Homepage/SearchResults.js:42]: i think you might have fixed https://github.com/covid-maps/covid-maps/issues/159

File: src/translations/hindi.js [hunk 2]
```diff
@@ -33,4 +33,6 @@ export default {
   select_location: "दुकान का स्थान चुनें",
   select_location_alert: "जारी रखने के लिए स्थान चुनें",
   closing_time_error: '"बंद करने का समय" खोलने के समय के बाद होना चाहिए',
+  form_submit_error: "दुर्भाग्य से हम आपका डेटा नहीं भेज सके। हम इसे जल्द से जल्द ठीक करने की कोशिश कर रहे हैं।",
```
Comment (arjun27) [inline src/translations/hindi.js:36]: love this translation

File: src/Homepage/index.js [hunk 3]
```diff
@@ -50,7 +53,8 @@ class Homepage extends React.Component {
     translations: PropTypes.object.isRequired,
     ipLocation: PropTypes.object,
     geoLocation: PropTypes.object,
-    setIPlocation: PropTypes.func.isRequired
+    setIPlocation: PropTypes.func.isRequired,
+    location: PropTypes.object.isRequired,
```
Comment (arjun27) [inline src/Homepage/index.js:57]: this is the router location, right? any change we could rename it to routerLocation? or maybe just add a comment?

File: src/Homepage/index.js [hunk 4]
```diff
@@ -62,25 +66,60 @@ class Homepage extends React.Component {
...
+      const queryParams = qs.parse(this.props.location.search, { ignoreQueryPrefix: true });
+      if ('submittedStore' in queryParams) {
+        const storeData = JSON.parse(atob(queryParams.submittedStore));
+        selectedLocation = { lat: storeData.Latitude, lng: storeData.Longitude }
+        selectedStoreName = storeData[FORM_FIELDS.STORE_NAME];
+      }
+
       this.setState({
-        results: this.formatResults(data, location),
+        results: this.formatResults(data, selectedLocation || location),
```
Comment (arjun27) [inline src/Homepage/index.js:87]: can you merge from master and update this? i updated the logic of where the `api.query` call is now made

File: src/Homepage/SearchResults.js (hunk 1 above)
Comment (Siddharth11) [reply src/Homepage/SearchResults.js:42]: i'm currently using the store name, should i switch to `storeId` ?

Comment (Siddharth11) [reply src/Homepage/SearchResults.js:42]: Do you send the store data in the form submission api response ?

Comment (arjun27) [reply src/Homepage/SearchResults.js:42]: Yes the submission response sends back store data. I think store name is ok for now

**Pull request merged**